docs/credentials.md

```diff
@@ -1,250 +1,525 @@
+
+You can set region for an individual service by specifying it as an
+argument when you create the service object. This take precedence over
+region set using any other method.
 
 ``` r
 svc <- paws::svc(
     config = list(
-        region = "us-east-1"
+        region = "us-east-2"
     )
 )
 ```
 
-In this example, `paws::svc` is a placeholder; use a specific service instead,
-such as `paws::s3`.
+In this example, `paws::svc` is a placeholder for an AWS service. Use a
+specific service instead, for example, `paws::s3`. Paws supports having
+multiple service objects with different regions.
+
+---
+
+
+# Set profile
+
+You can store different settings together under different profiles in the AWS
+shared credentials and config files.
 
-### Setting Region with R Environment Variable
+If you do not specify a profile to use, Paws will use the `default` profile.
 
-Use R to set the region with the following command:
+You can specify a different profile to use using the `AWS_PROFILE` environment
+variable:
 
 ``` r
-Sys.setenv(AWS_REGION = "us-east-1")
+Sys.setenv(
+    AWS_PROFILE = "my-profile"
+)
 ```
 
-## Setting Region with System Environment Variable
-
-On Mac or Linux you can use the command line to set the region with the
-following command:
+Or using the `profile` parameter to an individual service:
 
-``` bash
-export AWS_REGION="us-east-1"
+``` r
+svc <- paws::svc(
+    config = list(
+        credentials = list(
+            profile = "your profile"
+        )
+    )
+)
 ```
 
-Paws currently does not support setting the region using a system
-variable on Windows.
+In the AWS shared credentials, a separate profile looks like the following:
 
-### Setting Region with AWS Config File
+    [default]
+    aws_access_key_id=default AWS access key
+    aws_secret_access_key=default AWS secret key
 
-You can set the region using a config file in `~/.aws/config`. You can
-generate a config file using the AWS CLI’s `aws configure` command.
+    [my-profile]
+    aws_access_key_id=my-profile AWS access key
+    aws_secret_access_key=my-profile AWS secret key
 
-The config file should be in INI format and should look like:
+And in the AWS config file, it looks like the following. Note that profile
+names other than default begin with `profile`.
 
     [default]
     region=us-east-1
 
-By default, Paws will use the `default` profile. If you would rather use
-a different profile, you can set the preferred profile by following the
-instructions in the “Profile” section of this document.
+    [profile my-profile]
+    region=us-east-2
 
-Profiles are added to the config file by appending them to the file
-below the default:
+---
 
-    [default]
-    region=region-1
-    
-    [profile other_profile]
-    region=region-2
 
-## AWS Profile
+# Reference
 
-If you do not set your AWS profile, the package will use the `default`
-profile.
+Credentials and options can be set in the following locations. Paws will check
```

Comment:
We should add a header for this section with a name like
```
## Credential priority
```

Reply:
Added with links.